Source/Core/InputCommon/GCAdapter.cpp

```diff
@@ -37,6 +37,7 @@ static std::atomic<int> s_controller_payload_size = {0};
 static std::thread s_adapter_thread;
 static Common::Flag s_adapter_thread_running;
 
+static std::recursive_mutex s_init_mutex;
```

Comment:
A recursive_mutex is a lock that allows the same locking thread to acquire ownership multiple times. It's not a barrier for all other threads.